### PR TITLE
Add git completions

### PR DESCRIPTION
**Pull request by cstyles**
## Description

This PR adds completions for `git-merge-base`, `git-ls-files`, `git-describe`, and `git-worktree`.

Unfortunately, I didn't notice that @ken-tunc opened a PR less than a day ago that also adds completions for `git-worktree` (#5794) until I went to submit this PR. ¯\\_(ツ)_/¯

EDIT: So I compared our branches and here are the differences I found:
1. @ken-tunc's PR will use the currently checked out branch as the completion description which is useful.
2. Unfortunately, there are bugs with the above feature that break completion when the worktree has a detached HEAD or when the worktree's directory contains whitespace.
3. My PR allows users to complete both the path and the \<commit-ish\> arguments to the `add` subcommand. It also adds tags to the list of suggestions for the \<commit-ish\> argument.
4. @ken-tunc's PR is missing a few options (`--[no-]guess-remote` for the `add` subcommand and  `-f`/`--force` for the `move` subcommand).
5. @ken-tunc's PR adds a completion to the `help` command.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

**Comment by ken-tunc** [comment]
Thank you for finding differences between our branches!
This PR looks better than mine, so I closed my PR.

**Comment by cstyles** [comment]
If you figure out a way to fix the bugs with the "currently checked out branch as the completion description" thing, please open a PR because it'd be a nice feature to have. I spent some time looking into it but couldn't figure out an easy way to do it.

File: share/completions/git.fish [hunk 1]
```diff
@@ -1135,6 +1160,35 @@ complete -c git -n '__fish_git_using_command log' -l no-prefix
...
+complete -c git -n '__fish_git_using_command ls-files' -s m -l modified -d 'Show modified files in the output'
+complete -c git -n '__fish_git_using_command ls-files' -s o -l others -d 'Show other (i.e. untracked) files in the output'
+complete -c git -n '__fish_git_using_command ls-files' -s i -l ignored -d 'Show only ignored files in the output'
+complete -c git -n '__fish_git_using_command ls-files' -s s -l staged -d "Show staged contents' mode bits, object name and stage number in the output"
+complete -c git -n '__fish_git_using_command ls-files' -l directory -d 'If a whole directory is classified as "other", show just its name'
+complete -c git -n '__fish_git_using_command ls-files' -l no-empty-directory -d 'Do not list empty directories'
+complete -c git -n '__fish_git_using_command ls-files' -s u -l unmerged -d 'Show unmerged files in the output'
+complete -c git -n '__fish_git_using_command ls-files' -s k -l killed -d 'Show files on the filesystem that need to be removed for checkout-index to succeed'
+complete -c git -n '__fish_git_using_command ls-files' -s z -d '\0 line termination on output and do not quote filenames'
```
Comment (faho) [inline share/completions/git.fish]: `Use \0 delimiter` or similar?

File: share/completions/git.fish [hunk 2]
```diff
@@ -1135,6 +1160,35 @@ complete -c git -n '__fish_git_using_command log' -l no-prefix
...
+complete -c git -n '__fish_git_using_command ls-files' -s x -l exclude -d 'Skip untracked files matching pattern'
+complete -c git -n '__fish_git_using_command ls-files' -s X -l exclude-from -d 'Read exclude patterns from <file>; 1 per line'
+complete -c git -n '__fish_git_using_command ls-files' -l exclude-per-directory -d 'Read additional exclude patterns that apply only to the directory and its subdirectories in <file>'
+complete -c git -n '__fish_git_using_command ls-files' -l exclude-standard -d 'Add the standard Git exclusions'
+complete -c git -n '__fish_git_using_command ls-files' -l error-unmatch -d 'If any <file> does not appear in the index, treat this as an error'
+complete -c git -n '__fish_git_using_command ls-files' -l with-tree
+complete -c git -n '__fish_git_using_command ls-files' -s t -d 'Identifies the file status'
+complete -c git -n '__fish_git_using_command ls-files' -s v -d 'Similar to -t but use lowercase letters for files that are marked as assume unchanged'
```
Comment (faho) [inline share/completions/git.fish]: "Show file status, use lowercase letters for files assumed unchanged"?

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**